src/main/java/bio/terra/app/controller/DatasetsApiController.java

```diff
@@ -153,6 +154,36 @@ public ResponseEntity<DatasetModel> retrieveDataset(
     return ResponseEntity.ok(datasetService.retrieveDatasetModel(id, userRequest, include));
   }
 
+  @Override
+  public ResponseEntity<SnapshotBuilderDatasetModel> retrieveSnapshotBuilderDataset(UUID id) {
+    AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+    iamService.verifyAuthorization(
+        userRequest, IamResourceType.DATASET, id.toString(), IamAction.READ_DATASET);
+    DatasetModel datasetModel =
+        datasetService.retrieveDatasetModel(
+            id,
+            userRequest,
+            List.of(
+                DatasetRequestAccessIncludeModel.STORAGE,
+                DatasetRequestAccessIncludeModel.ACCESS_INFORMATION,
+                DatasetRequestAccessIncludeModel.PROPERTIES,
+                DatasetRequestAccessIncludeModel.SCHEMA,
+                DatasetRequestAccessIncludeModel.DATA_PROJECT));
+    SnapshotBuilderDatasetModel snapshotBuilderDatasetModel =
+        new SnapshotBuilderDatasetModel()
+            .createdDate(datasetModel.getCreatedDate())
+            .description(datasetModel.getDescription())
+            .accessInformation(datasetModel.getAccessInformation())
+            .name(datasetModel.getName())
+            .cloudPlatform(datasetModel.getCloudPlatform())
+            .id(datasetModel.getId())
+            .properties(datasetModel.getProperties())
+            .schema(datasetModel.getSchema());
+    // TODO: Fetch root concept from each concept table
+    // TODO: Figure out where to put the learn-more link
```

Comment:
TDR likes to keep its controllers lightweight, with their methods handling authorization where appropriate, delegating the heavy lifting to a service, then constructing the response.  So this block would be a better fit for a cohort builder service IMO.